### ISSUE TITLE
Speaker Diarization doesnt not work for async conversation transcription .

### ISSUE THREAD

**Issue by madCodes** [Azure-Samples/cognitive-services-speech-sdk#1212]
I have downloaded the "conversation transcription " quickstart and ran in "async" transcription mode for the sample audio file "katiesteve.wav" without registering the voice samples. I get the transcriptions as expected, but the userId I get in the result is unidentified for both the speakers. But as per the note in the link [here ](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/conversation-transcription), I should get "Speaker1" and "Speaker2" in the userId, no?

"User voice samples are optional. Without this input, the transcription will show different speakers, but shown as "Speaker1", "Speaker2", etc. instead of recognizing as pre-enrolled specific speaker names."

**Comment by sarahlume** [comment]
Speaker diarization for guest which doesn't have profile is disabled per our legal advice.
If you would like to turn on this feature, Add the following code
config.SetProperty("DifferentiateGuestSpeakers", "True");

**Comment by madCodes** [comment]
@sarahlume  Thank you, I have tried the "DifferentiateGuestSpeakers" setting as suggested and now, I see one user is getting UserId as "Guest_0" and the other as "Unidentified".  Is this expected? This may be enough to get around diarization for just 2 users though :)

**Comment by sarahlume** [comment]
Yes, It's expected. And if the second guest continue talking, he/she will also get recognized. With the current audio, the service didn't get enough audio to get the second guest identified

**Comment by jhakulin** [comment]
Closed as answered, let us know if there is still an issue.

**Comment by madCodes** [comment]
Thanks. FYI, I have observed that for longer audios, initial few transcriptions has one user as unidentified and later transcriptions for the same user as Guest_02 even for the "async" only mode. with RemoteConversationTranscriptionClient . Also, Is there a way we can hint the system that there are only two people in the conversation?

**Comment by msjunzh** [comment]
@madCodes It's expected as the algorithm will sometimes output "Unidentified" speaker when it's not sure about the guest tagging. The algorithm/model are trained targeting multiple dataset and optimized for overall experiences.
It will be great if you can grant & share the "longer audio" with us(great if there're transcriptions) to further investigation.

**Comment by sarahlume** [comment]
We do have some unofficial API to hint the system the count of people in the conversation.
Here is the C# example             
var connection = Connection.FromRecognizer(recognizer);
connection.SetMessageProperty("speech.config", "userSpecifiedMinAttendeeCount", $"\"{minSpeakerNumber}\"");
connection.SetMessageProperty("speech.config", "userSpecifiedMaxAttendeeCount", $"\"{maxSpeakerNumber}\"");

**Comment by madCodes** [comment]
@msjunzh Unfortunately I cannot share those audios. But I downloaded a sample conversation from the public domain and as mentioned in [#1216 ](https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/1216), it stops after few transcriptions. 
[Long_Weekends_Memberax2wk.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/7009809/Long_Weekends_Memberax2wk.zip)

**Comment by BohdanVovkotrub** [comment]
How to enable "DifferentiateGuestSpeakers" in NodeJS SDK ?
I try "speechConfig.setProperty("DifferentiateGuestSpeakers", true);"
but privSpeakerId is undefined in result.
